### DISCUSSION TITLE
How can i avoid the page being briefly displayed unstyled when loading it.

### DISCUSSION THREAD

**Discussion by eryilmazyasin** [vercel/next.js#59632] [Help]
### Summary

I created a new next js project which is version 14.0.4. I installed emotion/css for styling and MUI. When i refresh the page first MUI styles displays then my custom css styles displays. I want to override MUI styles but it displays every page refresh.  Here is my files below. I tried a lot of solution but could not fix my problem. 

https://github.com/vercel/next.js/assets/40694757/1b6a1eef-db49-425d-9169-d9531b8b3fba

<img width="1672" alt="Screenshot 2023-12-15 at 00 35 14" src="https://github.com/vercel/next.js/assets/40694757/d3eeb452-fc7b-4f86-b9fc-fd406b611e82">

<img width="1675" alt="Screenshot 2023-12-15 at 00 36 11" src="https://github.com/vercel/next.js/assets/40694757/9201f602-0d40-4468-89e1-4cc1c6248667">

**Comment by DevAbdoTolba** [comment]
I would suggest adding a loading screen animation that takes almost nothing just until the styling loads.
Another aproch is to install @emoition/cache

1. 

```bash
npm install @mui/material-nextjs @emotion/cache
```

2. Configure your layout file located at : app/layout.tsx, import the AppRouterCacheProvider and wrap all elements under the <body> with it:

```ts
import { AppRouterCacheProvider } from '@mui/material-nextjs/v14-appRouter';

export default function RootLayout (props) {
  const { children } = props;
  return (
    <html lang="en">
      <body>
        <AppRouterCacheProvider options={{ enableCssLayer: true }}>
          {children}
        </AppRouterCacheProvider>
      </body>
    </html>
  );
}
```

[The enableCssLayer: true option ensures that the styles generated by Material UI will be wrapped in a CSS @layer mui rule, which is overridden by anonymous layer styles when using Material UI with CSS modules, Tailwind CSS, or even plain CSS without using @layer](https://mui.com/material-ui/guides/nextjs/)

3. Configure your document file: Inside the pages/_document.tsx file, import documentGetInitialProps and use it as the Document’s getInitialProps. [Import DocumentHeadTags and render it inside the <Head>](https://mui.com/material-ui/guides/nextjs/)
4. [Configure your app file: Inside pages/_app.tsx, import the AppCacheProvider component and render it as the root element](https://mui.com/material-ui/guides/nextjs/)[1](https://mui.com/material-ui/guides/nextjs/).
[Please replace v14-appRouter with v13-appRouter if you are using Next.js v13](https://mui.com/material-ui/guides/nextjs/)[1](https://mui.com/material-ui/guides/nextjs/).

Remember, the order of styles in CSS matters. The last style rule defined for a specific selector takes precedence over earlier rules. [If you’re using a styling solution other than Emotion to customize Material UI components, setting enableCssLayer: true in the options prop ensures that the styles generated by Material UI will be wrapped in a CSS @layer mui rule, which is overridden by anonymous layer styles](https://mui.com/material-ui/guides/nextjs/)

I hope this helps! Let me know if you have any other questions. 😊
